compiler/rustc_middle/src/dep_graph/dep_node.rs

```diff
@@ -51,6 +53,12 @@
 //! than a zeroed out fingerprint. More generally speaking, it relieves the
 //! user of the `DepNode` API of having to know how to compute the expected
 //! fingerprint for a given set of node parameters.
+//!
+//! [^1]: For an overview of the dependency graph and its role in
+//!   the query system and incremental compilation, see
+//!   [the rustc-dev-guide](https://rustc-dev-guide.rust-lang.org/query.html).
+//!
+//! [`Fingerprint`]: rustc_data_structures::fingerprint::Fingerprint
```

Comment:
This link resolution is unnecessary since `Fingerprint` is in scope:

```suggestion
```

Reply:
I might have had an issue with this locally... Will check again, thanks.